### DISCUSSION TITLE
Image Updates - January 13, 2023

### DISCUSSION THREAD

**Discussion by navarr** [wardenenv/warden#574] [Announcements]
Images and their build process have been ported over from Den for all services excluding PHP.

Newly built and pushed services:

* MariaDB: 10.0, 10.1, 10.2, 10.3, 10.4, 10.5, 10.6, 10.7, 10.8, 10.9, and 10.10-rc (`MARIADB_VERSION`)
* Varnish: 6.0, 6.5, 6.6, 7.0, 7.1, 7.2 (`VARNISH_VERSION`)
* ElasticSearch: 5, 5.6, 6, 7.10, 7.12, 7.13, 7.14, 7.15, 7.16, 7.17 (`ELASTICSEARCH_VERSION`)
* OpenSearch: 1.1, 1.2, 1.3, 2.0, 2.1, 2.2, 2.3, 2.4 (not yet automatically configurable)
* MagePack: 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 2.10, 2.11 (`MAGEPACK_VERSION`)
* Nginx: 1.16, 1.17, 1.18, 1.19, 1.20, 1.21, 1.22, 1.23 (`NGINX_VERSION`)
* RabbitMQ: 3.7, 3.8, 3.9, 3.10, 3.11 (`RABBITMQ_VERSION`)
* Redis: 3.2, 4.0, 5.0, 6.0, 6.2, 7.0 (`REDIS_VERSION`)
* MySQL: 5.5, 5.6, 5.7, 8.0.28, 8.0 (not yet automatically configurable)

New versions of Mailhog and DNSMasq have also been pushed up.  The services are available via Docker Hub and the GitHub Container Registry.

**PHP Image**

* PHP has been updated so that you can specify `COMPOSER_VERSION` 1, 2, or 2.2 - the later being Composer 2's LTS version (currently required by Magento)

As always, please report any issues you come across!

**Comment by bap14** [comment]
What's the blocker on porting the PHP images?

**Reply by navarr** [reply]
No blocker.  Just working my way there.  I want to ensure backwards compatibility, so even once ported they will not be immediately accessible.

**Reply by navarr** [reply]
Well, that said, the recent Node issues in the Den PHP images have to be resolved before I port them in.  My original solution of baking them into the images by pulling them from the node-alpine images is not working great due to different alpine versions

**Reply by bap14** [reply]
Sounds good!

**Comment by navarr** [comment]
Original Post updated with an addition to the PHP image

**Comment by alexandrosk** [comment]
Updating warden doesn't let me use opensearch yet? as WARDEN_OPENSEARCH=1

Also is there a way to commit and fix the Documentation so we can update it as well, thanks :)

**Reply by navarr** [reply]
You'll have to use the develop branch as it hasn't been fully tested/released yet. To do that, you either have to checkout the repository and run Warden from there or run `warden install wardenenv/warden/warden --HEAD`

**Reply by navarr** [reply]
Documentation is at https://github.com/wardenenv/docs

**Reply by alexandrosk** [reply]
Yes I did this works perfect with open search thanks
